python/pyspark/tests/test_context.py

```diff
@@ -246,6 +251,16 @@ def test_startTime(self):
         with SparkContext() as sc:
             self.assertGreater(sc.startTime, 0)
 
+    def test_forbid_insecure_gateway(self):
+        # Fail immediately if you try to create a SparkContext
+        # with an insecure gateway
+        mock_insecure_gateway = MagicMock()
+        mock_insecure_gateway.gateway_parameters = MagicMock()
+        mock_insecure_gateway.gateway_parameters.auth_token = None
```

Comment:
@squito, I think it doesn't necessarily use mock .. I think it's okay to work around, for instnace by namedtuple:

```python
from collections import namedtuple

parameters = namedtuple('MockGatewayParameters', 'auth_token')(None)
mock_insecure_gateway = namedtuple('MockJavaGateway', 'gateway_parameters')(parameters)
```